### PR TITLE
more defensive refresh token

### PR DESCRIPTION
**Pull request by the-daniel-rothig**
### Context

https://trello.com/c/ANTTDi3U/299-bug-401-unauthorised-from-dfe-sign-in

### Changes proposed in this pull request

- hardcode timeout to just under 20 minutes
- handle refresh error case (by signing out)

### CODE REVIEW

File: src/ui/Startup.cs [hunk 1]
```diff
@@ -123,6 +122,12 @@ public void ConfigureServices(IServiceCollection services)
                                 // between cookie and access token is preserved
                                 x.ShouldRenew = true;
                             }
+                            else 
+                            {
+                                // could not refresh - log the user out
+                                _logger.LogWarning("Token refresh failed with message: " + response.ErrorDescription);
+                                x.RejectPrincipal();
```
[the remark this answers is not in the document]
Comment (the-daniel-rothig) [reply src/ui/Startup.cs]: No they get signed out + sent in to dfe sign in's login page

[the remark this answers is not in the document]
Comment (defong) [reply src/ui/Startup.cs]: You sure? 
Our code only governs our system and our cookies, they should still be sign in to dfe sign in.

They will be rejected by our system but they should still be logged in to dfe so they be bounce back to us.

[the remark this answers is not in the document]
Comment (the-daniel-rothig) [reply src/ui/Startup.cs]: as mentioned in meat-space: DfE Sign in currently prompts users to log in again even if they are still signed in on the system. But even if DfE Sign in would just bounce them back, they would do so with a new access_token so it's all good :+1:

File: src/ui/Startup.cs [hunk 2]
```diff
@@ -123,6 +122,12 @@ public void ConfigureServices(IServiceCollection services)
                                 // between cookie and access token is preserved
                                 x.ShouldRenew = true;
                             }
+                            else 
+                            {
+                                // could not refresh - log the user out
+                                _logger.LogWarning("Token refresh failed with message: " + response.ErrorDescription);
```
Comment (defong) [inline src/ui/Startup.cs]: Can we log the subject id as well

Comment (the-daniel-rothig) [reply src/ui/Startup.cs]: I don't think the subjectid is in scope at that point - if I understand right, we'd have to query the API to figure this out which I don't want to do

Comment (defong) [reply src/ui/Startup.cs]: Its in scope
```
var identity = (ClaimsIdentity) x.Principal.Identity;
                            var accessTokenClaim = identity.FindFirst("access_token");	                            var accessTokenClaim = identity.FindFirst("access_token");
```

**Pull request merged**